cookbooks/aws-parallelcluster-slurm/recipes/update_head_node.rb

```diff
@@ -50,6 +50,87 @@
   not_if { ::File.exist?(node['cluster']['previous_cluster_config_path']) && !are_queues_updated? }
 end
 
+ruby_block "replace slurm queue nodes" do
+  SLURM_POWER_SAVING_MAPPING = {
+    DRAIN: "POWER_DOWN_ASAP",
+    TERMINATE: "POWER_DOWN_FORCE",
+  }.freeze
+
+  def execute_command(command, timeout)
+    cmd = Mixlib::ShellOut.new(command, user: "root", timeout: timeout)
+    cmd.run_command
+    raise_command_error(command, cmd) if cmd.error?
+    cmd.stdout.strip
+  end
+
+  def raise_command_error(command, cmd)
+    Chef::Log.error("Error while executing command (#{command})")
+    raise "#{cmd.stderr.strip}"
+  end
+
+  def get_slurm_nodelist(queue)
+    command = "cat /opt/slurm/etc/pcluster/slurm_parallelcluster_#{queue}_partition.conf | grep NodeSet | cut -d' ' -f2 | cut -d'=' -f2"
```

Comment:
could you add an example of the expected output of this line?

Or what about using something like `cat /opt/slurm/etc/pcluster/slurm_parallelcluster_#{queue}_partition.conf | sed -n 's/.*NodeSet.*Nodes=\(.*\)/\1/p'` that is self explicative and is just a single sed command?

Reply:
added example and switched logic to use a sed